mpt_trie/src/trie_subsets.rs

```diff
@@ -21,13 +22,15 @@ use crate::{
 /// The output type of trie_subset operations.
 pub type SubsetTrieResult<T> = Result<T, SubsetTrieError>;
 
-/// Errors that may occur when creating a subset [`PartialTrie`].
+/// We can't construct the error until we unwind the stack from the error
+/// creation site. Once we unwind enough, we convert it to the actual error
+/// type.
+type SubTrieResultIntern<T> = Result<T, ()>;
```

Comment:
I don't think this is needed - one could return a `Result<(), ()>` at the (single) usage site, no?

Reply:
True, there is only one usage. I tend to honestly prefer using aliases and functions even when they are only used once. Even though it introduces more indirection, I find it gets the code closer to reading like English and I think the tradeoff is usually worth it.

In this case however, "Intern" doesn't really tell the reader that much more I guess and I think I agree that it's a bit worse, so yeah lets remove this.